runtime/lua/vim/ui/img/utils/screen.lua

```diff
@@ -0,0 +1,149 @@
+---@class vim.ui.img.utils.ScreenSize
+---@field width number in pixels (may be fractional)
+---@field height number in pixels (may be fractional)
+---@field columns integer
+---@field rows integer
+---@field cell_width number in pixels (may be fractional)
+---@field cell_height number in pixels (may be fractional)
+---@field scale number dpi
+
+---@class vim.ui.img.utils.Screen
+---@field private __def boolean has created necessary cdefs
+---@field private __size vim.ui.img.utils.ScreenSize|nil cached screen size
+local M = {
+  __def = false,
+  __size = nil,
+}
+
+vim.api.nvim_create_autocmd('VimResized', {
+  desc = 'Screen size has changed',
+  callback = function()
+    -- Clear our cache when the screen size changes
+    M.__size = nil
+  end,
+})
+
+---Convert an integer representing absolute pixels to a cell.
+---Rounds to the nearest integer.
+---@param x integer
+---@param y integer
+---@return integer x, integer y
+function M.pixels_to_cells(x, y)
+  local size = M.size()
+  return math.floor((x / size.cell_width) + 0.5), math.floor((y / size.cell_height) + 0.5)
+end
+
+---Convert an integer representing a cell to absolute pixels.
+---Rounds to the nearest integer.
+---@param x integer
+---@param y integer
+---@return integer x, integer y
+function M.cells_to_pixels(x, y)
+  local size = M.size()
+  return math.floor((x * size.cell_width) + 0.5), math.floor((y * size.cell_height) + 0.5)
+end
+
+---Determines the size of the terminal screen.
+---@return vim.ui.img.utils.ScreenSize
+function M.size()
+  local size = M.__size
+
+  if size then
+    return size
+  end
+
+  if vim.fn.has('unix') == 1 then
+    size = M.__posix_size()
+  end
+
+  M.__size = size or M.__default_size()
+
+  return M.__size
+end
+
+---@private
+---Determines the size of the terminal screen for POSIX systems.
+---@return vim.ui.img.utils.ScreenSize|nil
+function M.__posix_size()
+  ---@type vim.ui.img.utils.ScreenSize|nil
+  local size
+
+  -- On Linux/Android, BSD, MacOS, and Solaris we use
+  -- ioctl with TIOCGWINSZ to calculate the size.
```

Comment:
don't we have  `VimResize` event? or see `runtime/lua/vim/termcap.lua` , `TermResponse` (idk if that helps)

Reply:
`VimResized` lets you know of the resize, but we can only get the rows and columns from neovim, not the pixel dimensions, which we want to figure out how big a cell in the terminal is in relation to pixels to perform the conversions.